backend/director/agents/comparison.py

```
@@ -0,0 +1,129 @@
+                        future = executor.submit(
+                            self._run_video_generation, index, params
+                        )
+                        future.add_done_callback(
+                            lambda fut: self._update_videos_content(
+                                videos_content, fut.result()[0], fut.result()[1]
+                            )
+                        )
+                        futures.append(future)
+
+                    # Process completed tasks as they finish
+                    for future in concurrent.futures.as_completed(futures):
```

Ruff (0.8.2)

Loop control variable `future` not used within loop body

(B007)